src/main/kotlin/com/nikitahohulia/listeningplatform/repository/PostRepository.kt

```diff
@@ -0,0 +1,12 @@
+package com.nikitahohulia.listeningplatform.repository
+
+import com.nikitahohulia.listeningplatform.entity.Post
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.repository.MongoRepository
+import org.springframework.stereotype.Repository
+
+@Repository
+interface PostRepository : MongoRepository<Post, ObjectId> {
+
+    fun getPostsByCreatorIdOrderByCreatedAt(creatorId: ObjectId): List<Post>
```

Comment:
I'm glad that you managed to use Spring Data for your repositories. But let's try writing queries to mongo manually, using Criteria API. you can leave some of repositories as spring data repositories and rewrite some with criteria

Reply:
Done